electionleaflets/settings/base_lambda.py

```diff
@@ -52,3 +52,11 @@
 USE_X_FORWARDED_HOST = True
 
 setup_sentry()
+
+# importing test settings file if necessary (TODO chould be done better)
+if len(sys.argv) > 1 and sys.argv[1] in ["test", "harvest"]:
+    try:
+        from .testing import *  # noqa: F401,F403
+    except ImportError:
+        pass
+
```

Comment:
Follow up from the surgery.
Fundamentally, the thing we're trying to achieve is to use the settings in `electionleaflets.settings.testing` under test. I think we're doubling down on unhelpful hacks here. My proposed patch for how to set this up is:

```patch
diff --git a/.circleci/config.yml b/.circleci/config.yml
index 04003c1..9823c57 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -70,7 +70,7 @@ jobs:
       - run:
           name: Run tests
           command: |
-            pipenv run python manage.py test
+            pipenv run test
       - store_artifacts:
           path: test-results
           destination: test-results
diff --git a/Pipfile b/Pipfile
index f3ad52a..af3cce5 100644
--- a/Pipfile
+++ b/Pipfile
@@ -87,3 +87,4 @@ createcachetable = "python manage.py createcachetable"
 coveralls = "coveralls"
 migratedb = "python manage.py migrate --noinput"
 thumbs_sam_build = "bash -c 'cd thumbs && sam build'"
+test = "bash -c 'DJANGO_SETTINGS_MODULE=electionleaflets.settings.testing python manage.py test'"
diff --git a/electionleaflets/settings/base.py b/electionleaflets/settings/base.py
index 4a8310b..2b2b084 100644
--- a/electionleaflets/settings/base.py
+++ b/electionleaflets/settings/base.py
@@ -253,11 +253,3 @@ if not environ.get("DEPLOYMENT", None):
         ]
     except ImportError:
         pass
-
-
-# importing test settings file if necessary (TODO chould be done better)
-if len(sys.argv) > 1 and sys.argv[1] in ["test", "harvest"]:
-    try:
-        from .testing import *  # noqa: F401,F403
-    except ImportError:
-        pass
diff --git a/electionleaflets/settings/base_lambda.py b/electionleaflets/settings/base_lambda.py
index 3e65a41..06686f1 100644
--- a/electionleaflets/settings/base_lambda.py
+++ b/electionleaflets/settings/base_lambda.py
@@ -52,11 +52,3 @@ CSRF_TRUSTED_ORIGINS = [".electionleaflets.org"]
 USE_X_FORWARDED_HOST = True
 
 setup_sentry()
-
-# importing test settings file if necessary (TODO chould be done better)
-if len(sys.argv) > 1 and sys.argv[1] in ["test", "harvest"]:
-    try:
-        from .testing import *  # noqa: F401,F403
-    except ImportError:
-        pass
-
diff --git a/electionleaflets/settings/testing.py b/electionleaflets/settings/testing.py
index 10e6787..075da25 100644
--- a/electionleaflets/settings/testing.py
+++ b/electionleaflets/settings/testing.py
@@ -8,23 +8,4 @@ TEST_RUNNER = "django.test.runner.DiscoverRunner"
 
 MEDIA_ROOT = MEDIA_ROOT = root("test_media",)  # noqa: F405
 
-if os.environ.get("RUNNER") == "travis":
-    DATABASES = {
-        "default": {
-            "ENGINE": "django.contrib.gis.db.backends.postgis",
-            "NAME": "travis_ci_test",
-            "USER": "postgres",
-        }
-    }
-elif os.environ.get("RUNNER") == "circleci":
-    DATABASES = {
-        "default": {
-            "ENGINE": "django.contrib.gis.db.backends.postgis",
-            "NAME": "electionleaflets_test",
-            "USER": "postgres",
-            "HOST": "127.0.0.1",
-            "PORT": "5432",
-        }
-    }
-
 INSTALLED_APPS.append("aloe_django")  # noqa: F405
diff --git a/manage.py b/manage.py
index 8acd670..4cd190c 100755
--- a/manage.py
+++ b/manage.py
@@ -3,7 +3,7 @@ import os
 import sys
 
 if __name__ == "__main__":
-    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "electionleaflets.settings.base")
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "electionleaflets.settings")
 
     from django.core.management import execute_from_command_line
 

```

Doing it this way will also make it easier to swap unittest out for pytest/pytest-django in a future PR. I've also pushed a copy of that diff here: 
https://github.com/DemocracyClub/electionleaflets/commit/c9748d5a0264ada40661b5799c7cd64f1d3dd96c
and annotated it to explain what I'm doing at each stage and why.
Hopefully this helps not only do this more robustly, but also shed some light on how this works?

Reply:
Much clearer - esp how to add test settings in `pyproject.toml` - thank you.